### ISSUE TITLE
Overhaul project and replace inadequate tech (Beam, Dataflow, BigQuery)

### ISSUE THREAD

**Issue by darren-the** [darren-the/crypto-trading-v2#30]
This issue suggests a major overhaul to the current state of the project. There are currently several large problems that are hindering progress.

### Problem 1: Insufficient pipeline management tool
As of now, the pipeline s being developed using **Apache Beam** and **Google Cloud Dataflow**, however this has become increasingly problematic. The  library + technology focuses on managing the resources/infrastructure for processing data in "batch" and "streaming" pipelines (typically data is processed with a `map` function). Since candlestick data is processed sequentially (one element at a time, often depending on the results of previously processed elements), this presents several problems:
- It is impossible to distribute the work for a task across multiple workers. Instead, multiple workers can only be leveraged by having multiple separate tasks.
- Following the above point, having many separate and independant tasks is not how this pipeline should be handled as many tasks may depend on each other. Even if this was attempted, beam does not support efficiently branching and combining parallel tasks (for sequential data).

Basically, building a pipeline that processes sequential data, in parallel, with intertwining dependencies is not supported by beam.

### Problem 2: Lack of a proper dev environment and prod deployment process
In the current state, to quickly test a pipeline feature this can only be done by printing to a text file. Ideally, new features would be visualized in a chart to get a much more complete picture. Right now the steps to do this would be to:
1. push the changes to a branch
2. build the dataflow image on google cloud
3. start the pipeline on google cloud
4. once the pipelien is finished, open the chart on a node dev environment

This method doesn't make too much sense though for two reasons:
- Steps 1-3 takes about 10 minutes and this is just to get to the point where the pipeline job is started, no data has actually been processed yet. This is obviously becomes very impractical if someone wants to do testing over and over again.
- Another issue is the whole idea of "deploying" changes to google cloud and then switching to a local dev server to view the results. It's a bit strange that during development, half of the code is already _sort of_ deployed while the other half is on a dev server, it would make much more sense (and make the whole dev/testing process much more seamless) if everything happened on a dev server and was able to be ran purely in a single environment. This would also make deploying to production less confusing as well as we could deploy all the images at once.

**Comment by darren-the** [comment]
Reminder to document how these problems are being addressed

**Comment by darren-the** [comment]
### Briefly addressing problem 1

No publicly available pipeline management tool seems to fit the criteria that is needed, which is not surprising given how specific the requirements are. Thus, building a custom package appears to be the best alternative moving forward.

### Briefly addressing problem 2

Pipeline will be containerized and data will be stored on a local postgresql server both of which can be easily deployed to a serverless cloud platform.
